lib/util/grpc/state.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Util.Grpc.State do
+  use Agent
+
+  @moduledoc """
+  Keeps the configuration state for every Grcp client.
+
+  Example usage:
+
+    State.add_client(Client.new(name: "service1", endpoint: "localhost:9000", ...))
+    State.find_client("service1") => %Client{}
+
+  """
+
+  def start_link do
+    Agent.start_link(fn -> %{} end, name: __MODULE__)
+  end
+
+  def add_client(client) do
+    Agent.update(__MODULE__, fn state -> Map.put(state, client.name, client) end)
+  end
+
+  def find_client(name) do
+    Agent.get(__MODULE__, fn state -> Map.fetch!(state, name) end)
```

Comment:
Minor thing: `Agent.get/2` callback is executed internally by Agent process, and Map.fetch! raises unhandled exception in case of failure. Calling `find_client/1` with unknown will crash the State, not sure if that's by design?

Reply:
Good point. I'll have to fix it ✍️